### ISSUE TITLE
Proposition: `onStepError` callback

### ISSUE THREAD

**Issue by Odonno** [amiceli/vitest-cucumber#143]
The proposition is to provide the ability to configure an `onError` callback function to be executed once a step fails.

## Implementation details

If the `onError` property is defined, is should add a try/catch on each step and when a step fails, execute the callback in the catch block. The callback function parameter being at least the error thrown and the context of the test to interact with it if necessary.

Otherwise if the property is not set (null | undefined), no need to englobe the step function within a try/catch.

## Use cases

One principal use case is to print a [testing playground url](https://testing-library.com/docs/dom-testing-library/api-debugging/#screenlogtestingplaygroundurl) whenever an error occurs in a test.

Example:

```ts
const config = {
  // TODO : maybe add a check with an ENV variable to avoid using this callback in CI
  onError: (error, ctx) => {
    console.error(error);
    screen.logTestingPlayground();
  },
};

setVitestCucumberConfiguration(config);
```

**Comment by amiceli** [comment]
`onError` is a good idea, with `setVitestCucumberConfiguration` you did is useful.
I can do it if you want ;).

**Comment by amiceli** [comment]
Ok I will add it soon.

**Comment by amiceli** [comment]
I used `onTestFailed` to call `onStepError`. 
It works, but it's a little boring to test ^^.

I write this test : 

~~~typescript
describeFeature(feature, (f) => {
    f.Scenario('Simple scenario', (s) => {
        s.Given('I throw an error', () => {
            throw new Error('test')
        })
        s.Then('my onStepError is called', () => {
            expect(onStepError).toHaveBeenCalledWith({
                error: new Error('test'),
                step: feature.scenarii[0]?.steps[0],
                ctx: expect.any(Function),
            })
        })
    })
})
~~~

`onStepError` is called in `Then`, but tests still failed due to `throw new Error()` in `Given`.
I'm still working on it.

**Comment by amiceli** [comment]
I found a solution for tests, I'm finishing to implement `onStepError`.

**Comment by amiceli** [comment]
@Odonno Check PR, say me if this follows your proposition.

**Comment by amiceli** [comment]
Now available with `3.11.0` version.
